xds/internal/xdsclient/client_new.go

```diff
@@ -75,8 +75,12 @@ func newWithConfig(config *bootstrap.Config, watchExpiryTimeout time.Duration, i
 // Testing Only
 //
 // This function should ONLY be used for testing purposes.
-func NewWithConfigForTesting(config *bootstrap.Config, watchExpiryTimeout time.Duration) (XDSClient, error) {
-	cl, err := newWithConfig(config, watchExpiryTimeout, defaultIdleAuthorityDeleteTimeout)
+func NewWithConfigForTesting(config *bootstrap.Config, watchExpiryTimeout, authorityIdleTimeout time.Duration) (XDSClient, error) {
+	idleTimeout := defaultIdleAuthorityDeleteTimeout
+	if authorityIdleTimeout != time.Duration(0) {
```

Comment:
Fair point. Done.

Reply:
I don't think your change is correct, or it doesn't keep it consistent with previous commit. If you take 0 now as an arg, it sets it as 0, whereas previously it would set it as the default, which is 5 minutes?

Reply:
Most tests don't care about authority idle timeout, and they can continue to set `time.Duration(0)` instead of setting some default value. The tests which actually care about the idle authority timeout will set a small value anyways. So, I think this is correct. This API is a testing only API anyways, and none of the existing tests fail because of this change.

Reply:
Ok fair enough, it changed commit on commit though. LGTM.